### PR TITLE
Add "Connection Complete" page

### PR DESCRIPTION
**Pull request by aldo-expensify**
Can be tested with: https://github.com/Expensify/Web-Expensify/pull/41719

When the QBO authorization flow starts from the desktop app, we can't close the tab we opened. In this case, we want to show the user this new page so they manually go back to the Desktop app:

image

### Fixed Issues

$ https://github.com/Expensify/App/issues/40365
PROPOSAL:

### Tests

**For testing just the page being added directly:**

1. Navigate in an incognito window to https://dev.new.expensify.com:8082/connection-complete

**For testing the FULL flow, you need https://github.com/Expensify/Web-Expensify/pull/41719, then follow the steps below:**

In Web/IOS/Android:

1. Create workspace in NewDot
2. Go to the Workspace's settings and enable `Accounting` in `More Features`
3. Go to `Accounting` and start the flow to connect to QBO
4. Verify that in the end, the tab that opened for signing in into qbo closes automatically

In Desktop:

1. Create workspace in NewDot
2. Go to the Workspace's settings and enable `Accounting` in `More Features`
3. Go to `Accounting` and start the flow to connect to QBO
4. Verify that in the end, the tab redirects to this page:

image

- [x] Verify that no errors appear in the JS console

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I verified that similar component doesn't exist in the codebase
- [x] I verified that all props are defined accurately and each prop has a `/** comment above it */`
- [x] I verified that each file is named correctly
- [x] I verified that each component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
- [x] I verified that the only data being stored in component state is data necessary for rendering and nothing else
- [x] In component if we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
- [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
- [x] I verified that component internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
- [x] I verified that all JSX used for rendering exists in the render method
- [x] I verified that each component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/pages/ConnectionCompletePage.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+
+function ConnectionCompletePage() {
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    return (
+        <View style={styles.deeplinkWrapperContainer}>
+            <View style={styles.deeplinkWrapperMessage}>
+                <View style={styles.mb2}>
```
Comment (shawnborton) [inline src/pages/ConnectionCompletePage.tsx:19]: Maybe we can kill the margin bottom here - since it seems like there is quite a big gap between the image and text at the moment.

File: src/pages/ConnectionCompletePage.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                    <Icon
+                        width={variables.modalTopIconWidth}
```
Comment (shawnborton) [inline src/pages/ConnectionCompletePage.tsx:21]: Can you make sure we show this illustration at 272 x 188?

File: src/pages/ConnectionCompletePage.tsx (hunk 1 above)
Comment (aldo-expensify) [reply src/pages/ConnectionCompletePage.tsx:19]: In case it was making things look different, I updated the screenshot so you can see the whole screen 

image

Is the gap still too big?

Comment (shawnborton) [reply src/pages/ConnectionCompletePage.tsx:19]: I actually think maybe we need to fix the image size first, and then we can revisit the gap.

File: src/pages/ConnectionCompletePage.tsx (hunk 2 above)
Comment (aldo-expensify) [reply src/pages/ConnectionCompletePage.tsx:21]: Those sizes don't match any of the sizes we have in [variables](https://github.com/Expensify/App/blob/18694a48f1693dd9a3f0d45780432c7c2b2b5a60/src/styles/variables.ts#L138), I'll just hardcode here I guess

Comment (shawnborton) [reply src/pages/ConnectionCompletePage.tsx:21]: Works for me. Our illustrations are not all the same size, so I think that's okay.

Comment (aldo-expensify) [reply src/pages/ConnectionCompletePage.tsx:21]: Setting height:188 and width:272 it looks like this now: 
image

I think the space between the icon and the text got smaller because the height I had before was too much and was just filled with empty space

Comment (aldo-expensify) [reply src/pages/ConnectionCompletePage.tsx]: Updated

Comment (shawnborton) [reply src/pages/ConnectionCompletePage.tsx]: Exactly! So yeah, let's leave it like this then.

File: src/pages/ConnectionCompletePage.tsx (hunk 1 above)
Comment (shawnborton) [reply src/pages/ConnectionCompletePage.tsx:19]: Cool, no need to change the gap now that the image is the right size.

**Review by hayata-suenaga — APPROVED** [review]

**Review by hayata-suenaga — CHANGES_REQUESTED** [review]
sorry I clicked a wrong button. I meant to ask if we're showing the connection completion screen on web as well ask desktop?

**Comment by shawnborton** [comment]
Can't seem to login with the test builds above, but the screenshots look good to me at least!

**Comment by hayata-suenaga** [comment]
facing an issue with local testing. I messages Aldo 1:1

**Comment by aldo-expensify** [comment]
@shawnborton you should be able to see it by going here: https://40497.pr-testing.expensify.com/connection-complete

**Comment by aldo-expensify** [comment]
> I meant to ask if we're showing the connection completion screen on web as well ask desktop?

No, the connection complete show only be visible when you start the flow from the Desktop App. If you start the flow from web, the tab will close automatically and won't have time to see it.

**Comment by shawnborton** [comment]
Nice, looks great to me!

CleanShot 2024-04-23 at 08 06 03@2x

**Comment by aldo-expensify** [comment]
@hayata-suenaga is there something missing here to get this merged?

**Comment by hayata-suenaga** [comment]
sorry was trying to test locally, but ad hoc is working correctly. I'll approve the PR now.

**Review by hayata-suenaga — APPROVED** [review]

**Comment by hayata-suenaga** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by hayata-suenaga** [comment]
@aldo-expensify 

I was still testing on Desktop (my bad I shouldn't have approved and checked the boxes before testing is complete 😓 )

I believe the PR is working, but I was wondering how you ran web and desktop local build at the same time? Desktop build seems to be erroring whenever I ran it along with web 🤔 


This is the screen recording (at the last part, you can see some issues I faced with webpack)

https://drive.google.com/file/d/1Cq2L6iFD-AxN8p45zbyC_XxOvS8YhyIt/view?usp=sharing

**Comment by aldo-expensify** [comment]
> I believe the PR is working, but I was wondering how you ran web and desktop local build at the same time? Desktop build seems to be erroring whenever I ran it along with web 🤔

hmm I haven't seen a problem when running both, I'll check in main now that the changes have been merged

The errors I see in your video look unrelated to the changes here.
image

**Comment by aldo-expensify** [comment]
I tested the following and didn't see errors:

1. Run the desktop app and open the app in web too (chrome)
2. Login in both with the same account
3. Using the desktop App, created a workspace and connected to QBO
4. The QBO flow opened in a tab in chrome
5. Finished the authentication without errors

**Comment by hayata-suenaga** [comment]
let me try one more time

**Comment by hayata-suenaga** [comment]
I pulled the main and tested now the changes from this PR were merged. I still see the same issue

> Login in both with the same account

I think this is the different step. I didn't log into web new Expensify in advance to connecting to QBO.

Let me try again

**Comment by hayata-suenaga** [comment]
it seems that I cannot even open a New Expensify page on web (Incognito mode) when I run `npm run desktop`

I'll try troubleshooting when I have time, but because other issues don't require testing on desktop, I'll do that when I have time.

Screenshot 2024-04-30 at 11 23 26 PM

**Comment by aldo-expensify** [comment]
> it seems that I cannot even open a New Expensify page on web (Incognito mode) when I run npm run desktop

maybe a dumb question, but you are running `npm run web` in a different terminal, right?

**Comment by hayata-suenaga** [comment]
> maybe a dumb question, but you are running npm run web in a different terminal, right?

I tried both: not running and running `npm run web` in another terminal

both didn't work

**Comment by aldo-expensify** [comment]
hmm maybe you can show me in a google meet what you are doing, I have no idea what the problem may be.

**Comment by hayata-suenaga** [comment]
Iet's proritize QBO lease first and then we can circle back to this. thank you so much for offering help
